src/main/java/org/openlmis/stockmanagement/web/ValidSourceDestinationSearchParams.java

```diff
@@ -48,12 +49,12 @@ public ValidSourceDestinationSearchParams(MultiValueMap<String, String> queryMap
    *
    * @return UUID value of program id or null if params doesn't contain this param.
    */
-  public UUID getProgramId() {
-    if (!queryParams.containsKey(PROGRAM_ID)) {
+  public Set<UUID> getProgramId() {
+    if (!queryParams.containsKey(PROGRAM_ID) || queryParams.getFirst(PROGRAM_ID) == null) {
```

Comment:
I changed the implementation of this method. However, in case of an empty list we have to return null [due to the way the application works](https://github.com/OpenLMIS/openlmis-stockmanagement/blob/master/src/main/java/org/openlmis/stockmanagement/service/SourceDestinationBaseService.java#L144).